src/Security/AuthenticationRequestFilter.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace SilverStripe\Security;
+
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Control\RequestFilter;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\Session;
+use SilverStripe\ORM\DataModel;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\ORM\ValidationException;
+
+class AuthenticationRequestFilter implements RequestFilter, IdentityStore
+{
```

Comment:
I would feel a lot more comfortable about splitting this into two classes; AuthenticationRequestFilter should implement only RequestFilter, but should take a core IdentityStore as a dependency (e.g. IdentityStoreChain, as it contains chained nested identitystore services).